### PR TITLE
DDF-1509: Prevent selecting multiple records when they overlap (in the OpenLayers 2D map)

### PR DESCRIPTION
**Pull request by jrnorth**
@kcwire is the hero
@EIrwin 
@stustison 
@pklinef

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/193)

### CODE REVIEW

**Comment by bdeining** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/193)

**Comment by bdeining** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/193)

**Comment by EIrwin** [comment]
lgtm :thumbsup: 


---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/193)

File: catalog/ui/search-ui/standard/src/main/webapp/js/controllers/openlayers.controller.js [hunk 1]
```diff
@@ -115,18 +115,17 @@ define(['underscore',
...
-                        feature = controller.mapViewer.forEachFeatureAtPixel(event.pixel,
-                            function(feature) {
-                                controller.trigger('click:left', feature);
-                            }, this, function(testLayer) {
-                                if (testLayer === layer) {
-                                    return true;
-                                }
-                            });
-                    });
+                    controller.mapViewer.forEachFeatureAtPixel(event.pixel,
```
Comment (pklinef) [inline catalog/ui/search-ui/standard/src/main/webapp/js/controllers/openlayers.controller.js]: Won't this still select multiple metacards if they are in the same pixel/location?  If so, you might have to keep state in the closure about if a metacard has been found yet and only trigger if none have been found yet.

Comment (jrnorth) [inline catalog/ui/search-ui/standard/src/main/webapp/js/controllers/openlayers.controller.js]: After we detect a metacard and trigger the click event, we return true which will stop feature detection: http://openlayers.org/en/v3.2.1/apidoc/ol.Map.html#forEachFeatureAtPixel

**Comment by millerw8** [comment]
Reviewed 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/193)

**Pull request merged**